drgn_tools/rds.py

```diff
@@ -875,6 +977,41 @@ def rds_info_verbose(
             except AttributeError:
                 rcq_vector = "N/A"
 
+            if ensure_mlx_core_debuginfo(
+                prog, dev_name
+            ) and ensure_mlx_ib_debuginfo(prog, dev_name):
```

Comment:
The result of these functions does not change at all throughout this inner loop (for each connection on the device) yet we're calling these ensure debuginfo functions every single iteration.

Let's move them out of the inner loop, at the very least. There can be a lot of connections on a single device and that will multiple to many unnecessary calls.